synapse/handlers/presence.py

```diff
@@ -609,8 +605,15 @@ async def set_state(
             PresenceState.BUSY,
         )
 
+        busy_presence_enabled = (
+            await self.hs.get_datastores().main.get_feature_enabled(
+                target_user.to_string(), "msc3026"
+            )
+            or self.hs.config.experimental.msc3026_enabled
+        )
```

Comment:
Please check the homeserver config *first* so that we can avoid database hits in some situations:

```suggestion
        busy_presence_enabled = (
            await self.store.get_feature_enabled(
                target_user.to_string(), "msc3026"
            )
            or self.hs.config.experimental.msc3026_enabled
        )
```

Just to double check this means that the feature can be enabled per user or for all users? Would it make sense to put the check for `self.hs.config.experimental.msc3026_enabled` into `get_feature_enabled` so that all the logic is in a single spot?

This comment applies to pretty much every spot we call `get_feature_enabled`

Reply:
The feature can be enabled per user or for all features - this was a request from Erik (I know it diverges from what we had discussed in the issue.) I do think it makes sense to put all the logic checking whether it's enabled into one spot, and I've made that change.